system_tests/bigquery.py

```diff
@@ -673,8 +670,58 @@ def test_create_table_nested_schema(self):
         retry_403(dataset.create)()
         self.to_delete.append(dataset)
 
-        table = dataset.table(table_name, schema=_load_json_schema())
+        schema = _load_json_schema()
+        table = dataset.table(table_name, schema=schema)
         table.create()
         self.to_delete.insert(0, table)
         self.assertTrue(table.exists())
         self.assertEqual(table.name, table_name)
+
+        to_insert = []
+        # Data is in "JSON Lines" format, see http://jsonlines.org/
+        json_filename = os.path.join(WHERE, 'bigquery_test_data.jsonl')
+        with open(json_filename) as rows_file:
+            for line in rows_file:
+                mapping = json.loads(line)
+                to_insert.append(
+                    tuple(mapping[field.name] for field in schema))
+
+        errors = table.insert_data(to_insert)
+        self.assertEqual(len(errors), 0)
+
+        retry = RetryResult(_has_rows, max_tries=8)
+        fetched = retry(self._fetch_single_page)(table)
+        self.assertEqual(len(fetched), len(to_insert))
+
+        for found, expected in zip(sorted(fetched), sorted(to_insert)):
+            self.assertEqual(found[0], expected[0])            # Name
```

Comment:
I asserted that the lengths were equal above (line 694):  why would I assert a specific length, when that is driven by the data file, and might change if @tswast provides updates?